tests/trainer/test_fsdp_checkpoint.py

```diff
@@ -890,3 +892,183 @@ def test_cleanup_sharded_checkpoints(
         non_elastic_file_list = {save_filename.format(rank=rank) for rank in range(dist.get_world_size())}
         file_list = elastic_file_list if using_torch_2() else non_elastic_file_list
         assert set(os.listdir(full_path_ckpt_dir)) == file_list
+
+
+@pytest.mark.gpu
+@world_size(2)
+@pytest.mark.parametrize('weights_only', [False, True])
+@pytest.mark.parametrize('optimizer', ['adam', 'adamw'])
+@pytest.mark.parametrize('state_dict_type', ['sharded'])
+@pytest.mark.parametrize('precision', ['amp_bf16', 'amp_fp16'])
+@pytest.mark.parametrize('use_remote', [pytest.param(True, marks=pytest.mark.remote), False])
+@pytest.mark.parametrize('autoresume', [True, False])
+@pytest.mark.parametrize('planner', [None, 'rename'])
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.0'), reason='requires PyTorch 2.0 or higher')
+@pytest.mark.filterwarnings(r'ignore:TypedStorage is deprecated.:UserWarning')
+@pytest.mark.filterwarnings(r'ignore:.*metrics are not saved with sharded state dict.*:UserWarning')
+def test_fsdp_planner(
```

Comment:
Can you simplify this a bit since I don't think we need such a large matrix? I don't think a lot of these actually interact with loaders

In particular, I would remove:
- optimizer (adamw)
- precision (just bf16)
- use_remote (False)
- autoresume (False)
- dont need a test where planner = None?

and only keep weights only?

Reply:
Isn't it better to test more combinations?

The ones I think we definitely need are:
Planner = None vs not is useful because we don't want to make changes where None breaks.
I've had to make implementation changes for keep weights only vs not